domain/application/state/unit.go

```diff
@@ -559,6 +561,89 @@ func (st *State) InsertMigratingCAASUnits(ctx context.Context, appUUID coreappli
 	})
 }
 
+// SetRunningAgentBinaryVersion sets the running agent binary version for the
+// provided unit uuid. Any previously set values for this unit uuid will
+// be overwritten by this call.
+//
+// The following errors can be expected:
+// - [errors.UnitNotFound] if the unit does not exist.
+// - [coreerrors.NotSupported] if the architecture is not known to the database.
+func (st *State) SetRunningAgentBinaryVersion(ctx context.Context, uuid string, version coreagentbinary.Version) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	type ArchitectureMap struct {
```

Comment:
This doesn't need to be capitalised.
It's fine for now as it copies what was done for machine agent version. We'll sort out a common approach.